internal/server_control/server_control.go

```diff
@@ -34,10 +34,12 @@ type (
 		Start(context.Context) error
 		Stop(context.Context) error
 		Install(context.Context, string) error
+
+		State(context.Context) mine_state.State
```

Comment:
We always use verbs for function naming. Could we rename this function to GetState or something similar?